File: src/reset.js
'use strict';
// node:sqlite je v Node.js označený ako experimentálny - upozornenie do konzoly netreba.
const _emit = process.emitWarning;
process.emitWarning = (w, ...r) => { if (String(w).includes('SQLite')) return; _emit(w, ...r); };
// Zmaze databazu aj vygenerovanu postu a znovu naplni demo data.
const fs = require('fs');
const path = require('path');
const cfg = require('./config');

for (const f of [cfg.dbPath, cfg.dbPath + '-wal', cfg.dbPath + '-shm']) {
  if (fs.existsSync(f)) fs.unlinkSync(f);
}
const mailDir = path.join(path.dirname(cfg.dbPath), 'mail');
if (fs.existsSync(mailDir)) fs.rmSync(mailDir, { recursive: true, force: true });

const { migrate } = require('./db');
const { seedIfEmpty } = require('./seed');
migrate();
seedIfEmpty();
console.log('Databáza vyresetovaná a naplnená demo dátami.');

File: backend/app/reset.py
"""Zmaze DB a naplni odznova - mirror src/reset.js. Spusti: python -m app.reset"""
from __future__ import annotations

from . import config
from . import db
from .seed import seed_if_empty


def main() -> None:
    db.close()
    for suffix in ("", "-wal", "-shm"):
        p = config.DB_PATH.with_name(config.DB_PATH.name + suffix)
        if p.exists():
            p.unlink()
    db.reopen()
    db.migrate()
    seed_if_empty()
    print(f"DB obnovená a naplnená demo dátami: {config.DB_PATH}")


if __name__ == "__main__":
    main()
